### PR TITLE
end to end tests

### PR DESCRIPTION
**Pull request by slaskawi**
This Pull Request introduces e2e tests for self-signed certs.

### CODE REVIEW

**Comment by slaskawi** [comment]
Testing Prow 

/hold

File: test/library/dynamic_resources.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func NoErrorSkipNotExisting(t *testing.T, err error) {
+	if !k8serrors.IsNotFound(err) {
+		require.NoError(t, err)
+	}
+}
+
+func do(do doFunc, t *testing.T, assetFunc func(name string) ([]byte, error), filename string, kubeClient kubernetes.Interface, dynamicKubeClient dynamic.Interface) {
+	b, err := assetFunc(filename)
```
Comment (slaskawi) [inline test/library/dynamic_resources.go:36]: BTW, the implementation is based on https://gist.github.com/pytimer/0ad436972a073bb37b8b6b8b474520fc

File: test/e2e/cert_manager_deployment_test.go [hunk 2]
```diff
@@ -1,27 +1,53 @@
...
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "cluster_issuer.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.DeleteFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "cluster_issuer.yaml"), kubeClient, dynamicKubeConfig)
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "issuer.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.DeleteFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "issuer.yaml"), kubeClient, dynamicKubeConfig)
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "certificate.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "certificate.yaml"), kubeClient, dynamicKubeConfig)
+
+	err := wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+		secret, err := kubeClient.CoreV1().Secrets("sandbox").Get(context.TODO(), "root-secret", metav1.GetOptions{})
```
Comment (s-urbaniak) [inline test/e2e/cert_manager_deployment_test.go:41]: Let's wire those context.TODO()s here and elsewhere, since this is net-new code.

Comment (slaskawi) [reply test/e2e/cert_manager_deployment_test.go]: Done - I converted this part into a struct and implemented this in a more OO way.

**Comment by s-urbaniak** [comment]
let's merge this in so we have a first pass for e2e tests. i'll submit a follow-up PR if i see a better solution, thank you @slaskawi !

**Pull request merged**